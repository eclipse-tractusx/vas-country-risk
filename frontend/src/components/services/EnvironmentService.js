/********************************************************************************
 * Copyright (c) 2022,2024 BMW Group AG
 * Copyright (c) 2022,2024 Contributors to the Eclipse Foundation
 *
 * See the NOTICE file(s) distributed with this work for additional
 * information regarding copyright ownership.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Apache License, Version 2.0 which is available at
 * https://www.apache.org/licenses/LICENSE-2.0.
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 ********************************************************************************/
const LOCAL_SERVICES_FRONTEND =
  window.ENV?.REACT_APP_PORTAL_FRONTEND?.valueOf();
const REACT_APP_COUNTRY_RISK_API =
  window.ENV?.REACT_APP_COUNTRY_RISK_API?.valueOf();
const LOCAL_SERVICES_BACKEND = window.ENV?.REACT_APP_PORTAL_BACKEND?.valueOf();
const LOCAL_SERVICES_CENTRALIDP = window.ENV?.REACT_APP_AUTH_URL?.valueOf();
const LOCAL_COUNTRY_RISK_CLIENT_ID =
  window.ENV?.REACT_APP_COUNTRY_RISK_CLIENT?.valueOf();

export const getHostname = () => window.location.hostname;

export const isLocal = () => getHostname() === "localhost";

export const getApiBase = () =>
  isLocal()
    ? LOCAL_SERVICES_BACKEND
    : window.location.origin.replace("portal", "portal-backend");

export const getAssetBase = () =>
  `${isLocal() ? LOCAL_SERVICES_FRONTEND : ""}/assets`;

export const getCentralIdp = () => {
  return LOCAL_SERVICES_CENTRALIDP;
};

export const getPortalLink = () => {
  return LOCAL_SERVICES_FRONTEND;
};
export const getCountryRiskApi = () => {
  return REACT_APP_COUNTRY_RISK_API;
};

export const getLogoutLink = () => {
  return document.location.origin + "/logout";
};

export const getAboutLink = () => {
  return LOCAL_SERVICES_FRONTEND + "/about";
};

export const getCountryRiskAppId = () => {
  return LOCAL_COUNTRY_RISK_CLIENT_ID
    ? LOCAL_COUNTRY_RISK_CLIENT_ID
    : "country_risk_client";
};

export const getClientId = () => "Cl2-CX-Portal";

export const getCountryRiskClientId = () => "Cl16-CX-CRisk";

export const getBpdmId = () => "Cl7-CX-BPDM";

export const getClientIdSemantic = () => "Cl3-CX-Semantic";

export const getClientIdDigitalTwin = () => "Cl4-CX-DigitalTwin";

const EnvironmentService = {
  isLocal,
  getHostname,
  getApiBase,
  getAssetBase,
  getCentralIdp,
  getClientId,
  getClientIdSemantic,
  getClientIdDigitalTwin,
  getCountryRiskClientId,
  getCountryRiskAppId,
  getCountryRiskApi,
  getBpdmId,
};

export default EnvironmentService;
