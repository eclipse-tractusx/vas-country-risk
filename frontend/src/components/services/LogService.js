/********************************************************************************
* Copyright (c) 2022,2024 BMW Group AG
* Copyright (c) 2022,2024 Contributors to the Eclipse Foundation
*
* See the NOTICE file(s) distributed with this work for additional
* information regarding copyright ownership.
*
* This program and the accompanying materials are made available under the
* terms of the Apache License, Version 2.0 which is available at
* https://www.apache.org/licenses/LICENSE-2.0.
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
* License for the specific language governing permissions and limitations
* under the License.
*
* SPDX-License-Identifier: Apache-2.0
********************************************************************************/
const LogLevel = {
  SEVERE: "SEVERE",
  ERROR: " ERROR",
  WARN: "  WARN",
  INFO: "  INFO",
  DEBUG: " DEBUG",
};

export const logtime = () => new Date().toISOString().substring(11, 19);

export const log = (level, message) =>
  console.log(`${logtime()} ${level} ${message}`);

export const info = (message) => log(LogLevel.INFO, message);
export const warn = (message) => log(LogLevel.WARN, message);
export const error = (message) => log(LogLevel.ERROR, message);

const LogService = {
  LogLevel,
  logtime,
  log,
  info,
  warn,
  error,
};

export default LogService;
