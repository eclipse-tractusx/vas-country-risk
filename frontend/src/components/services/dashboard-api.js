/********************************************************************************
* Copyright (c) 2022,2024 BMW Group AG
* Copyright (c) 2022,2024 Contributors to the Eclipse Foundation
*
* See the NOTICE file(s) distributed with this work for additional
* information regarding copyright ownership.
*
* This program and the accompanying materials are made available under the
* terms of the Apache License, Version 2.0 which is available at
* https://www.apache.org/licenses/LICENSE-2.0.
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
* License for the specific language governing permissions and limitations
* under the License.
*
* SPDX-License-Identifier: Apache-2.0
********************************************************************************/
/* eslint-disable no-console */
import axios from "axios";
import  { getCountryRiskApi } from "./EnvironmentService"

// Actions
export function getAll(ratingsArray, years, token, customerUser, gates) {
  const ratingsParam = Array.isArray(ratingsArray)
    ? ratingsArray.map((rating) => JSON.stringify(rating))
    : [];

  return axios
    .get(getCountryRiskApi() + process.env.REACT_APP_DASHBOARD_URL, {
      params: {
        // Use the serialized and encoded ratingsParam
        "ratings[]": ratingsParam,
        year: years,
        name: customerUser.name,
        email: customerUser.email,
        companyName: customerUser.companyName,
        gate: gates,
      },
      headers: { Authorization: `Bearer ${token}` },
    })
    .then((res) => res.data)
    .catch((err) => err);
}

export function getWorldMapInfo(
  ratingsArray,
  years,
  token,
  customerUser,
  gates
) {
  const ratingsParam = Array.isArray(ratingsArray)
    ? ratingsArray.map((rating) => JSON.stringify(rating))
    : [];

  return axios
    .get(getCountryRiskApi() + process.env.REACT_APP_DASHBOARD_WOLRD_MAP_URL, {
      params: {
        // Use the serialized and encoded ratingsParam
        "ratings[]": ratingsParam,
        year: years,
        name: customerUser.name,
        email: customerUser.email,
        companyName: customerUser.companyName,
        gate: gates,
      },
      headers: { Authorization: `Bearer ${token}` },
    })
    .then((res) => res.data)
    .catch((err) => err);
}
