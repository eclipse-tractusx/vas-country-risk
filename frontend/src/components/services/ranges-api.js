/********************************************************************************
* Copyright (c) 2022,2024 BMW Group AG
* Copyright (c) 2022,2024 Contributors to the Eclipse Foundation
*
* See the NOTICE file(s) distributed with this work for additional
* information regarding copyright ownership.
*
* This program and the accompanying materials are made available under the
* terms of the Apache License, Version 2.0 which is available at
* https://www.apache.org/licenses/LICENSE-2.0.
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
* License for the specific language governing permissions and limitations
* under the License.
*
* SPDX-License-Identifier: Apache-2.0
********************************************************************************/
/* eslint-disable no-console */
import axios from "axios";
import { Range } from "../model/Range";
import { getCountryRiskApi } from "./EnvironmentService";

// Actions
export function getAllRanges(token, customerUser) {
  return axios
    .get(getCountryRiskApi()+process.env.REACT_APP_GET_RANGES, {
      params: {
        name: customerUser.name,
        email: customerUser.email,
        companyName: customerUser.companyName,
      },
      headers: { Authorization: `Bearer ${token}` },
    })
    .then((res) => res.data)
    .catch((err) => err);
}

export function sendValues(rangesList, customerUser, token) {
  const rangeDTOS = [];
  rangesList.forEach((element) => {
    let range = "";
    if (element[0] === 0) {
      range = new Range("Min", element[1], "Min Range");
    } else if (element[1] === 100) {
      range = new Range("Max", element[1], "Max Range");
    } else {
      range = new Range("Between", element[1], "BetWeen Range");
    }
    rangeDTOS.push(range);
  });

  return axios({
    method: "post",
    url: getCountryRiskApi()+process.env.REACT_APP_SAVE_RANGES,
    data: rangeDTOS,
    params: {
      name: customerUser.name,
      email: customerUser.email,
      companyName: customerUser.companyName,
    },

    headers: { Authorization: `Bearer ${token}` },
  });
}
