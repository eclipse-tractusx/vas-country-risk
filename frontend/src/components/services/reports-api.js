/********************************************************************************
* Copyright (c) 2022,2024 BMW Group AG
* Copyright (c) 2022,2024 Contributors to the Eclipse Foundation
*
* See the NOTICE file(s) distributed with this work for additional
* information regarding copyright ownership.
*
* This program and the accompanying materials are made available under the
* terms of the Apache License, Version 2.0 which is available at
* https://www.apache.org/licenses/LICENSE-2.0.
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
* License for the specific language governing permissions and limitations
* under the License.
*
* SPDX-License-Identifier: Apache-2.0
********************************************************************************/
/* eslint-disable no-console */
import axios from "axios";
import { getCountryRiskApi } from "./EnvironmentService";

//Get Reports By User
export function getReportsByCompanyUser(token, customerUser) {
  return axios
    .get(getCountryRiskApi() + process.env.REACT_APP_GET_REPORTS_BY_USER, {
      params: {
        name: customerUser.name,
        email: customerUser.email,
        companyName: customerUser.companyName,
      },
      headers: { Authorization: `Bearer ${token}` },
    })
    .then((res) => res.data)
    .catch((err) => []);
}

export function getReportValuesByReport(token, report, customerUser) {
  return axios
    .get(
      getCountryRiskApi() + process.env.REACT_APP_GET_REPORT_VALUES_BY_REPORT,
      {
        params: {
          reportName: report.reportName || "",
          companyUserName: report.companyUserName || "",
          company: report.company || "",
          type: report.type || "",
          id: report.id || "",
          name: customerUser.name,
          email: customerUser.email,
          companyName: customerUser.companyName,
        },
        headers: { Authorization: `Bearer ${token}` },
      }
    )
    .then((res) => res.data)
    .catch((err) => []);
}

//Save Reports By User
export function saveReports(token, customerUser, report) {
  return axios({
    method: "post",
    url: getCountryRiskApi()+process.env.REACT_APP_SAVE_REPORTS,
    data: report,
    params: {
      name: customerUser.name,
      email: customerUser.email,
      companyName: customerUser.companyName,
    },

    headers: { Authorization: `Bearer ${token}` },
  });
}

export function shareReports(token, customerUser, report) {
  return axios({
    method: "post",
    url:getCountryRiskApi()+ process.env.REACT_APP_SHARE_REPORTS,
    data: report,
    params: {
      name: customerUser.name,
      email: customerUser.email,
      companyName: customerUser.companyName,
    },

    headers: { Authorization: `Bearer ${token}` },
  });
}

//Update Reports By User
export function updateReports(token, customerUser, report) {
  return axios({
    method: "put",
    url:getCountryRiskApi()+ process.env.REACT_APP_UPDATE_REPORTS,
    data: report,
    params: {
      name: customerUser.name,
      email: customerUser.email,
      companyName: customerUser.companyName,
    },

    headers: { Authorization: `Bearer ${token}` },
  });
}

export function deleteReport(token, customerUser, reportId) {
  return axios({
    method: "delete",
    url:
      getCountryRiskApi() +
      process.env.REACT_APP_DELETE_REPORTS +
      `/${reportId}`,
    params: {
      name: customerUser.name,
      email: customerUser.email,
      companyName: customerUser.companyName,
    },

    headers: { Authorization: `Bearer ${token}` },
  });
}
