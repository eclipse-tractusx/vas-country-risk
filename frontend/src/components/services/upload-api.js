/********************************************************************************
 * Copyright (c) 2022,2024 BMW Group AG
 * Copyright (c) 2022,2024 Contributors to the Eclipse Foundation
 *
 * See the NOTICE file(s) distributed with this work for additional
 * information regarding copyright ownership.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Apache License, Version 2.0 which is available at
 * https://www.apache.org/licenses/LICENSE-2.0.
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 ********************************************************************************/

import axios from "axios";
import { getCountryRiskApi } from "./EnvironmentService";
import UserService from "./UserService";

// Uploads a CSV file to the server.
export function uploadCsvFile(file, ratingName, year, type, customerUser) {
  const url = getCountryRiskApi() + process.env.REACT_APP_UPLOAD_FILE;
  const formData = new FormData();
  formData.append("file", file);

  return axios.post(url, formData, {
    params: {
      name: customerUser.name,
      email: customerUser.email,
      companyName: customerUser.companyName,
    },
    headers: {
      "Content-Type": "multipart/form-data",
      Authorization: `Bearer ${UserService.getToken()}`,
      ratingName: ratingName,
      year: year,
      type: type,
    },
  });
}
