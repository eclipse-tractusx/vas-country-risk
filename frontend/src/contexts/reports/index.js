/********************************************************************************
* Copyright (c) 2022,2024 BMW Group AG
* Copyright (c) 2022,2024 Contributors to the Eclipse Foundation
*
* See the NOTICE file(s) distributed with this work for additional
* information regarding copyright ownership.
*
* This program and the accompanying materials are made available under the
* terms of the Apache License, Version 2.0 which is available at
* https://www.apache.org/licenses/LICENSE-2.0.
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
* License for the specific language governing permissions and limitations
* under the License.
*
* SPDX-License-Identifier: Apache-2.0
********************************************************************************/
import React, { createContext, useEffect, useState, useContext } from "react";
import { CompanyUserContext } from "../companyuser";
import { getReportValuesByReport } from "../../components/services/reports-api";
import UserService from "../../components/services/UserService";

const ReportContext = createContext({});

const ReportProvider = ({ children, updatedReport }) => {
  const [reportValuesContext, setReportValuesContext] = useState(
    updatedReport || []
  );

  const { companyUser } = useContext(CompanyUserContext);

  const [report, setReport] = useState("");

  useEffect(() => {
    getReportValuesByReport(UserService.getToken(), report, companyUser).then(
      (response) => {
        setReportValuesContext(response);
      }
    );
  }, [report]);

  const updateReport = (report) => {
    setReport(report);
  };

  return (
    <ReportContext.Provider
      value={{ report, reportValuesContext, updateReport }}
    >
      {children}
    </ReportContext.Provider>
  );
};

export { ReportContext, ReportProvider };
