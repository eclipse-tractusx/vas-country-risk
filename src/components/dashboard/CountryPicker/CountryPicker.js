/********************************************************************************
 * Copyright (c) 2022,2024 BMW Group AG
 * Copyright (c) 2022,2024 Contributors to the Eclipse Foundation
 *
 * See the NOTICE file(s) distributed with this work for additional
 * information regarding copyright ownership.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Apache License, Version 2.0 which is available at
 * https://www.apache.org/licenses/LICENSE-2.0.
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 ********************************************************************************/
import React, { useState, useEffect, useContext } from "react";
import "./styles.scss";
import Box from "@mui/material/Box";
import { TextField } from "@mui/material";
import Autocomplete from "@mui/material/Autocomplete";
import { getCountryByUser } from "../../services/country-api";
import UserService from "../../services/UserService";
import { CountryContext } from "../../../contexts/country";
import { CompanyUserContext } from "../../../contexts/companyuser";

const CountryPicker = () => {
  //Const with Countries relative to the user and BP
  const [Countries, setCountries] = useState("");

  //Context to save the current selected country
  const { countryS, updateCountry } = useContext(CountryContext);

  const { companyUser } = useContext(CompanyUserContext);

  //Call to get all the Countries relative to a User and BP
  useEffect(() => {
    getCountryByUser(UserService.getToken(), companyUser).then((response) => {
      setCountries(response);
    });
  }, []);

  //Handler to get current selected value on Autocomplete component
  const handleChange = (event, newValue) => {
    if (newValue === null) {
      updateCountry("none");
    } else {
      updateCountry(newValue);
    }
  };

  return (
    <Autocomplete
      className="autocomplete"
      fullWidth
      variant="filled"
      size="small"
      noOptionsText={"No country found"}
      onChange={handleChange}
      options={Countries || []}
      autoHighlight
      freeSolo={false}
      value={countryS || []}
      getOptionLabel={(option) => option.country || ""}
      renderOption={(props, option) => (
        <Box
          component="li"
          sx={{ "& > img": { mr: 2, flexShrink: 0 }, fontSize: 13 }}
          {...props}
        >
          <img
            loading="lazy"
            width="20"
            src={`https://flagcdn.com/w20/${option.iso2.toLowerCase()}.png`}
            srcSet={`https://flagcdn.com/w40/${option.iso2.toLowerCase()}.png 2x`}
          />
          {option.country} ({option.iso2})
        </Box>
      )}
      renderInput={(params) => (
        <TextField
          {...params}
          label="Select a country"
          InputProps={{
            ...params.InputProps,
          }}
          sx={{
            background: "white",
            borderRadius: "10px",
          }}
        />
      )}
    />
  );
};

export default CountryPicker;
