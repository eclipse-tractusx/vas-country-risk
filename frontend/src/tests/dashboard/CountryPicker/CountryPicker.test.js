/********************************************************************************
* Copyright (c) 2022,2024 BMW Group AG
* Copyright (c) 2022,2024 Contributors to the Eclipse Foundation
*
* See the NOTICE file(s) distributed with this work for additional
* information regarding copyright ownership.
*
* This program and the accompanying materials are made available under the
* terms of the Apache License, Version 2.0 which is available at
* https://www.apache.org/licenses/LICENSE-2.0.
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
* License for the specific language governing permissions and limitations
* under the License.
*
* SPDX-License-Identifier: Apache-2.0
********************************************************************************/
import { render, act, fireEvent, screen } from "@testing-library/react";
import CountryPicker from "../../../components/dashboard/CountryPicker/CountryPicker";
import { test } from "@jest/globals";
import { CountryProvider } from "../../../contexts/country";

import { getCountryByUser } from "../../../components/services/country-api";

import React from "react";
import renderer from "react-test-renderer";

const countryData = [
  {
    id: 0,
    country: "Germany",
    iso3: "DEU",
    iso2: "DE",
    continent: "Europe",
    latitude: "-2.9814344",
    longitude: "23.8222636",
    totalBpn: 11,
  },
];

jest.mock("../../../components/services/country-api", () => ({
  getCountryByUser: jest.fn().mockReturnValue(countryData),
}));

test("CountryPicker snapshot test", () => {
  const component = renderer.create(<CountryPicker />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("handleChange updates countryS context correctly", async () => {
  getCountryByUser.mockImplementation(() => Promise.resolve(countryData));

  await act(async () => {
    render(
      <CountryProvider>
        <CountryPicker />
      </CountryProvider>
    );
  });

  const autocomplete = screen.getByLabelText("Select a country");

  act(() => {
    fireEvent.change(autocomplete, { target: { value: "Germany" } });
    fireEvent.keyDown(autocomplete, { key: "ArrowDown" });
    fireEvent.keyDown(autocomplete, { key: "Enter" });
  });
});

test("handleChange updates countryS context with null", async () => {
  getCountryByUser.mockImplementation(() => Promise.resolve(countryData));

  await act(async () => {
    render(
      <CountryProvider>
        <CountryPicker />
      </CountryProvider>
    );
  });

  const autocomplete = screen.getByLabelText("Select a country");
  const clearBox = screen.getByTitle("Clear");
  act(() => {
    fireEvent.change(autocomplete, { target: { value: "Germany" } });
    fireEvent.keyDown(autocomplete, { key: "ArrowDown" });
    fireEvent.keyDown(autocomplete, { key: "Enter" });
    fireEvent.click(clearBox);
  });
});
