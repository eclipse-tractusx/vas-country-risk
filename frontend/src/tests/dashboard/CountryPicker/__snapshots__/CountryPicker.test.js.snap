// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryPicker snapshot test 1`] = `
<div
  aria-owns={null}
  className="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon autocomplete css-12szb0g-MuiAutocomplete-root"
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  variant="filled"
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor=":r0:"
      id=":r0:-label"
    >
      Select a country
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-invalid={false}
        autoCapitalize="none"
        autoComplete="off"
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r0:"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        readOnly={false}
        required={false}
        role="combobox"
        spellCheck="false"
        type="text"
        value=""
      />
      <div
        className="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
      >
        <button
          aria-label="Clear"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          title="Clear"
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
        <button
          aria-label="Open"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          title="Open"
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </button>
      </div>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            Select a country
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
